pkg/api/core/v1/client/apps_test.go

```diff
@@ -0,0 +1,61 @@
+package client_test
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/epinio/epinio/helpers/tracelog"
+	"github.com/epinio/epinio/pkg/api/core/v1/client"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Client Apps unit tests", func() {
+	var epinioClient *client.Client
+	var statusCode int
+	var responseBody string
+
+	BeforeEach(func() {
+		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(statusCode)
```

Comment:
```
When Ginkgo runs a spec it runs through all the BeforeEach closures that appear in that spec's hierarchy from the outer-most to the inner-most.
```
from here: https://onsi.github.io/ginkgo/#organizing-specs-with-container-nodes

If you want to DRY the code this way, I think you need `JustBeforeEach` which allows you to create the `srv` object after you set the `statusCode` and still keep it at the top level block.

Reply:
It works in both cases but it's cleaner with the `JustBeforeEach`, probably because the `statusCode` value is not fetched during the creation of the server but when the handlerFunc is executed.

I'll check the failing tests and add the fix!